src/base64Utils.js

```diff
@@ -4,33 +4,38 @@
  * SPDX-License-Identifier: MIT
  */
 
+#if SINGLE_FILE == 2 && !WASM_ASYNC_COMPILATION
+#error SINGLE_FILE == 2 does not work with synchronous WASM compilation (because DecompressionStream is async by nature)
+#endif
+
 #if POLYFILL && (ENVIRONMENT_MAY_BE_SHELL || (ENVIRONMENT_MAY_BE_NODE && MIN_NODE_VERSION < 160000))
 #include "polyfill/atob.js"
 #endif
 
 // Converts a string of base64 into a byte array (Uint8Array).
 function intArrayFromBase64(s) {
-#if ENVIRONMENT_MAY_BE_NODE
+#if ENVIRONMENT_MAY_BE_NODE && SINGLE_FILE != 2
   if (typeof ENVIRONMENT_IS_NODE != 'undefined' && ENVIRONMENT_IS_NODE) {
     var buf = Buffer.from(s, 'base64');
     return new Uint8Array(buf.buffer, buf.byteOffset, buf.length);
   }
 #endif
-
   var decoded = atob(s);
   var bytes = new Uint8Array(decoded.length);
   for (var i = 0 ; i < decoded.length ; ++i) {
     bytes[i] = decoded.charCodeAt(i);
   }
+#if SINGLE_FILE == 1 
   return bytes;
+#else
+  return new Response(new Blob([bytes]).stream().pipeThrough(new DecompressionStream("gzip"))).arrayBuffer();
```

Comment:
Could we continue to allow the above nodejs optimization that avoid atob?

Reply:
Oh yeah sure, only in nodejs though

Reply:
One way to do that i think would be to have `intArrayFromBase64 continue to have its existing well-defined meaning and we could an outer `gunzipByteArray` function?

Reply:
Any chance this code can be made async? If so you can use `fetch` to decode base64:

```
export async function parse_base64(data: string, data_type = 'octet-binary'): Promise<Uint8Array> {
    // Parse base64 using a trick from https://stackoverflow.com/a/54123275/2854284
    const response = await fetch(`data:application/${data_type};base64,${data}`)
    if (!response.ok) {
        throw new Error(`Could not parse base64: ${response.status}`)
    }
    return new Uint8Array(await response.arrayBuffer())
}
```

Reply:
Not easily I fear, since we need to continue to support synchronous compilation of the wasm file

Reply:
I think it might be cleaner to layer this decompression on top of the base64 utilities.  I.e. have the caller of `tryParseAsDataURI` take care of decompressing the restul?